courier/courier_dispatcher.go

```diff
@@ -48,10 +47,10 @@ func (c *courier) DispatchQueue(ctx context.Context) error {
 		return err
 	}
 
-	ttl := c.deps.CourierConfig(ctx).CourierMessageTTL()
+	maxRetries := c.deps.CourierConfig(ctx).CourierMessageRetries()
 
 	for k, msg := range messages {
-		if time.Now().After(msg.CreatedAt.Add(ttl)) {
+		if msg.SendCount >= maxRetries {
```

Comment:
How can we test that emails won't get send that reach the maximum retry count?

Reply:
That is the existing test. Since messages that have the `Abandoned` state are no longer queued, they won't be sent out. And they receive that state if the condition `msg.SendCount > maxRetries` is true.

I would like to have a mock SMTP server here, so we can test this case, because currently we are relying on the fact that no SMTP server is running at baz.com which is not ideal. WDYT about that?